code/firmware/rosco_m68k_firmware/stage2/main2.c

```diff
@@ -140,12 +154,18 @@ noreturn void lmain() {
 #ifdef WITH_SPLASH
     if (stage1_have_xosera) {
         intro_end();
-        MC_DELAY_MSEC_10(POST_SPLASH_DELAY_MSEC10);
+        if (!skip_splash)
+        {
+            MC_DELAY_MSEC_10(POST_SPLASH_DELAY_MSEC10);
+        }
 #endif
 #ifdef WITH_BOOT_MENU
-        if (handle_boot_menu()) {
+        boot_menu_res menu_res = handle_boot_menu();
+        if (menu_res == RES_LOAD_OK) {
             goto have_kernel;
-        };
+        }
+        if (menu_res == RES_UART_LOADER)
+            goto start_uart_loader;
```

Comment:
I think we'd need to skip this on MAME and just jump to the quick load (which I know is a pain)? Currently the build is failing for MAME firmware because this label isn't defined 😢

Reply:
(Thinking more about it, it's probably not worth building the menu for MAME since no Xosera yet, but maybe worth keeping in for debugging purposes etc 🤔 ).

Reply:
Okay...it needs to build at least. 🙂

Reply:
ifdefs added